OnlineTools/Utils/QueryStringParameterExtensions.cs

```diff
@@ -63,7 +63,7 @@ public async ValueTask UpdateQueryString<T>(T component, bool reloadPage = true)
                     continue;
 
                 var value = property.GetValue(component);
-                if (value is null)
+                if (value is null || (value.GetType() == typeof(string) && string.IsNullOrWhiteSpace(value.ToString())))
```

Comment:
- I don't think we should skip whitespace strings, only null or empty
- The check looks complicated. You can simplify it by using the `is` operator

```suggestion
                if (value is null || (value is string strValue && string.IsNullOrEmpty(strValue)))
```

Reply:
The reason I want to preserve non-empty strings is because it would be valid in some cases. In the following URL, the spaces are important.

````
https://tools.meziantou.net/regex-tester?Pattern=%5Cs%2B&Text=%20%20%20%20&Options=512
````

If it were something important, we could add an attribute to indicates properties where spaces are important. But I don't think there is a value to add complexity to this project :)